classic_cp/N_Queens/N_Queens.jl

```diff
@@ -0,0 +1,168 @@
+using SeaPearl
+
+struct OutputDataQueens
+    nb_sols ::Int
+    indices ::Matrix{Int}
+end
+
+"""
+    model_queens(board_size::Int; benchmark=false, variableSelection=SeaPearl.MinDomainVariableSelection{false}(), valueSelection=SeaPearl.BasicHeuristic())
+
+return the SeaPearl model for to the N-Queens problem, using SeaPearl.MinDomainVariableSelection heuristique
+and  SeaPearl.AllDifferent (without solving it)
+
+# Arguments
+- `board_size::Int`: dimension of the board
+- 'variableSelection': SeaPearl variable selection. By default: SeaPearl.MinDomainVariableSelection{false}()
+- 'valueSelection': SeaPearl value selection. By default: =SeaPearl.BasicHeuristic()
+"""
+function model_queens(board_size::Int; benchmark=false, variableSelection=SeaPearl.MinDomainVariableSelection{false}(), valueSelection=SeaPearl.BasicHeuristic())
+    trailer = SeaPearl.Trailer()
+    model = SeaPearl.CPModel(trailer)
+
+    rows = Vector{SeaPearl.AbstractIntVar}(undef, board_size)
+    for i = 1:board_size
+        rows[i] = SeaPearl.IntVar(1, board_size, "row_"*string(i), trailer)
+        SeaPearl.addVariable!(model, rows[i]; branchable=true)
+    end
+
+    rows_plus = Vector{SeaPearl.AbstractIntVar}(undef, board_size)
+    for i = 1:board_size
+        rows_plus[i] = SeaPearl.IntVarViewOffset(rows[i], i, rows[i].id*"+"*string(i))
+    end
+
+    rows_minus = Vector{SeaPearl.AbstractIntVar}(undef, board_size)
+    for i = 1:board_size
+        rows_minus[i] = SeaPearl.IntVarViewOffset(rows[i], -i, rows[i].id*"-"*string(i))
+    end
+
+    push!(model.constraints, SeaPearl.AllDifferent(rows, trailer))
+    push!(model.constraints, SeaPearl.AllDifferent(rows_plus, trailer))
+    push!(model.constraints, SeaPearl.AllDifferent(rows_minus, trailer))
+
+    return model
+end
+
+"""
+    solve_queens(board_size::Int; benchmark=false, variableSelection=SeaPearl.MinDomainVariableSelection{false}(), valueSelection=SeaPearl.BasicHeuristic())
+
+Solve the SeaPearl model for to the N-Queens problem, using SeaPearl.MinDomainVariableSelection heuristique
+and  SeaPearl.AllDifferent, and the function model_queens
+
+# Arguments
+- `board_size::Int`: dimension of the board
+- 'variableSelection': SeaPearl variable selection. By default: SeaPearl.MinDomainVariableSelection{false}()
+- 'valueSelection': SeaPearl value selection. By default: =SeaPearl.BasicHeuristic()
+"""
+function solve_queens(board_size::Int; benchmark=false, variableSelection=SeaPearl.MinDomainVariableSelection{false}(), valueSelection=SeaPearl.BasicHeuristic())
+    model = model_queens(board_size::Int; benchmark=false, variableSelection=SeaPearl.MinDomainVariableSelection{false}(), valueSelection=SeaPearl.BasicHeuristic())
+    status = @time SeaPearl.solve!(model; variableHeuristic=variableSelection, valueSelection=valueSelection)
+    return model
+end
+
```

Comment:
It could be useful to overload `solve_queens` with a function taking an existent model.

Reply:
`variableSelection` and `valueSelection` aren't defined in the new `solve_queens`.